src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2569,52 +2569,24 @@ bool SessionImpl::cancelDownloadMetadata(const TorrentID &id)
 
 void SessionImpl::increaseTorrentsQueuePos(const QList<TorrentID> &ids)
 {
-    using ElementType = std::pair<int, const TorrentImpl *>;
-    std::priority_queue<ElementType
-        , std::vector<ElementType>
-        , std::greater<ElementType>> torrentQueue;
-
-    // Sort torrents by queue position
-    for (const TorrentID &id : ids)
-    {
-        const TorrentImpl *torrent = m_torrents.value(id);
-        if (!torrent) continue;
-        if (const int position = torrent->queuePosition(); position >= 0)
-            torrentQueue.emplace(position, torrent);
-    }
+    QList<TorrentImpl *> queuedTorrents = getQueuedTorrentsByID(ids);
+    std::ranges::sort(queuedTorrents, std::greater<>(), &TorrentImpl::queuePosition);
```

Comment:
@Chocobo1 
We can even sort list within `getQueuedTorrentsByID()` and traverse it in reverse direction when needed.